elements/lisk-blocks/src/data_access/storage_access.ts

```diff
@@ -0,0 +1,281 @@
+		fromHeight: number,
+		toHeight: number,
+		tx?: StorageTransaction,
+	): Promise<BlockJSON[]> {
+		const blocks = await this._storage.entities.Block.get(
+			{ height_gte: fromHeight, height_lte: toHeight },
+			{ extended: true },
+			tx,
+		);
+
+		return blocks;
+	}
+
+	public async getTempBlocks(tx: StorageTransaction): Promise<TempBlock[]> {
+		const tempBlocks = await this._storage.entities.TempBlock.get(
+			{},
+			{ sort: 'height:asc' },
+			tx,
+		);
+
+		return tempBlocks;
+	}
+
+	public async isTempBlockEmpty(): Promise<boolean> {
+		const isEmpty = await this._storage.entities.TempBlock.isEmpty();
+
+		return isEmpty;
+	}
+
+	public clearTempBlocks(): void {
+		this._storage.entities.TempBlock.truncate();
+	}
+
+	public async getLatestBlock(): Promise<BlockJSON> {
+		const [lastBlock] = await this._storage.entities.Block.get(
+			{},
+			{ sort: 'height:desc', limit: 1, extended: true },
+		);
+
+		return lastBlock;
+	}
+
+	public async getFirstBlockIdWithInterval(
+		height: number,
+		interval: number,
+	): Promise<Array<Partial<BlockJSON>>> {
+		const rows: Array<Partial<
+			BlockJSON
+		>> = await this._storage.entities.Block.getFirstBlockIdOfLastRounds({
+			height,
+			numberOfRounds: 5,
+			numberOfDelegates: interval,
+		});
+
+		return rows;
+	}
+
+	public async isBlockPersisted(blockId: string): Promise<boolean> {
+		const isPersisted = await this._storage.entities.Block.isPersisted({
+			blockId,
+		});
+
+		return isPersisted;
+	}
+
+	public async getAccountsByPublicKey(
+		arrayOfPublicKeys: string[],
+		tx?: StorageTransaction,
+	): Promise<Account[]> {
+		const accounts = await this._storage.entities.Account.get(
+			{ publicKey_in: arrayOfPublicKeys },
+			{},
+			tx,
+		);
+
+		return accounts;
+	}
+
+	public async getAccountsByAddress(
+		arrayOfAddresses: string[],
+		tx?: StorageTransaction,
+	): Promise<Account[]> {
+		const accounts = await this._storage.entities.Account.get(
+			{ address_in: arrayOfAddresses },
+			{},
+			tx,
+		);
+
+		return accounts;
+	}
+
+	public async getDelegateAccounts(
+		tx?: StorageTransaction,
+	): Promise<Account[]> {
+		const accounts = await this._storage.entities.Account.get(
+			{ isDelegate: true },
+			{ limit: 101, sort: ['voteWeight:desc', 'publicKey:asc'] },
+			tx,
+		);
+
+		return accounts;
+	}
+
+	public async getActiveDelegatesByLimit(
+		limit: number,
+		tx?: StorageTransaction,
+	): Promise<ReadonlyArray<RoundDelegates>> {
+		const delegateLists = await this._storage.entities.RoundDelegates.get(
+			{},
+			{
+				sort: 'round:desc',
+				limit,
+			},
+			tx,
+		);
+
+		return delegateLists;
+	}
+
+	public async getActiveDelegatesForRound(
+		round: number,
+		tx?: StorageTransaction,
+	): Promise<ReadonlyArray<string>> {
+		const delegatePublicKeys = await this._storage.entities.RoundDelegates.getActiveDelegatesForRound(
+			round,
+			tx,
+		);
+
+		return delegatePublicKeys;
+	}
+
+	public async getTransactionsByIDs(
+		arrayOfTransactionIds: string[],
```

Comment:
```suggestion
		arrayOfTransactionIds: ReadonlyArray<string>,
```
In general if we are not mutating, let's keep it `Readonly`

Reply:
Using `ReadonlyArray<string>` produces a type error